### PR TITLE
Rfc/3154 full line and column location of methods

### LINKED ISSUE #3154: Make full line and column location of methods available in templates
# Feature request

Hi, I'm working on an open source clone detection tool (will be released in march, hopefully). In the tool there is an own implementation of searching methods in php files and resolving docBlocks belonging to these methods. 

Now I've found out, phpDocumentor exists. 😄 I like the speed and the accurancy of phpDocumentor. Now I'm implementing a XML template in order to replace my own implementation from above with a call to phpDocumentor.

phpDocumentor reports the starting line of methods, it found. My feature request it to also make the starting column, ending line and ending column of methods available as variables in templates. This would allow to extract the method contents from the phpDocumentor report.

I guess this could be done by extending `phpDocumentor\Descriptor\Builder\Reflector\MethodAssembler::mapReflectorToDescriptor` to something like this:
```php
protected function mapReflectorToDescriptor(Method $reflector, MethodDescriptor $descriptor): void
{
    ...
    $descriptor->setLine($reflector->getLocation()->getLineNumber());
    $descriptor->setColumn($reflector->getLocation()->getColumnNumber());
    $descriptor->setEndLine($reflector->getEndLocation()->getLineNumber());
    $descriptor->setEndColumn($reflector->getEndLocation()->getColumnNumber());
    ...
}
```

At the moment, when doing this, the starting line is resolved properly. The ending line and the column numbers are defaulting to `0`. I guess, this could be solved by adding the appropriate attributes to the options of PHP-Parser lexer (see https://github.com/nikic/PHP-Parser/blob/master/doc/component/Lexer.markdown).

This feature would be non-breaking, as it only extends but does not changes existing features.

What do you think about this? The only downside of this I can think of is, that there could be a small performance loose, when PHP-Parser has to resolve more attributes. Maybe this could be solved by making the PHP-Parser lexer attributes configurable via phpDocumentor config.

If nothing is against this, I'd like to implement this.

### PR DESCRIPTION
**Pull request by LeoVie**
This fixes #3154

### CODE REVIEW

**Comment by LeoVie** [comment]
Fixed code style. This should work now.

File: src/phpDocumentor/Descriptor/Builder/Reflector/MethodAssembler.php [hunk 1]
```diff
@@ -82,6 +82,9 @@ protected function mapReflectorToDescriptor(Method $reflector, MethodDescriptor
         $descriptor->setAbstract($reflector->isAbstract());
         $descriptor->setStatic($reflector->isStatic());
         $descriptor->setLine($reflector->getLocation()->getLineNumber());
+        $descriptor->setColumn($reflector->getLocation()->getColumnNumber());
```
[the remark this answers is not in the document]
Comment (LeoVie) [reply src/phpDocumentor/Descriptor/Builder/Reflector/MethodAssembler.php:85]: Yes, this seems to be cleaner. Let me change this.

[the remark this answers is not in the document]
Comment (LeoVie) [reply src/phpDocumentor/Descriptor/Builder/Reflector/MethodAssembler.php]: I've removed the `setLine` method and added the getter and setter methods for `startLocation` and `endLocation`. Also I deprecated the `getLine` method
Of course, there are more changes now in this pull request, but mostly they are nearly the same in the different assemblers.

**Comment by LeoVie** [comment]
The failing checks seem to be unrelated to the changes in this pull request.

**Comment by jaapio** [comment]
Thanks, I will have a look at what is going on with the tests.

**Pull request merged**